### PR TITLE
Restrictions on Script Public Keys with standardness rules

### LINKED ISSUE #53: Restrictions on Funding Change Script Public Keys
Do we want any restrictions on what change Script Public Keys are allowed to be?

In the current state of the specification (see #51) both parties pay equal fees for the funding transaction so I'm currently leaning towards starting out with the restriction that specifying the length of the SPK requires only one byte (i.e. the var_int for the spk length takes up only one byte of the transaction).

In the future we will likely move to a model where each participant pays the fees for their own inputs and outputs rather than sharing this burden, at which point perhaps this restriction should be lifted.

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #53 

standardness rules taken from: https://github.com/lightningnetwork/lightning-rfc/pull/672

### CODE REVIEW

**Review by nkohen — APPROVED** [review]

File: Protocol.md [hunk 1]
```diff
@@ -228,10 +234,25 @@ The recipient:
...
 
+### Script Pubkey Standardness Definition
+
+ For a script pub key to be valid it must be in one of the following forms:
+
+    1. `OP_DUP` `OP_HASH160` `20` 20-bytes `OP_EQUALVERIFY` `OP_CHECKSIG` (pay to pubkey hash), OR
+    2. `OP_HASH160` `20` 20-bytes `OP_EQUAL` (pay to script hash), OR
+    3. `OP_0` `20` 20-bytes (version 0 pay to witness pubkey hash), OR
+    4. `OP_0` `32` 32-bytes (version 0 pay to witness script hash), OR
+    5. `OP_1` through `OP_16` inclusive, followed by a single push of 2 to 40 bytes
```
Comment (ariard) [inline Protocol.md:277]: Are those rules scoping funding inputs script pubkey ? Note, non-Taproot and v1+ witness program spends aren't standards and will be rejected by the current Bitcoin Core policy (see `AreInputsStandard` in `src/policy/policy.cpp`). 

Unless we fully verify inputs pubkeys and witnesses against standardness rules, a malicious counterparty should be able to fail propagation of the funding tx...

File: Protocol.md [hunk 2]
```diff
@@ -228,10 +234,25 @@ The recipient:
...
+       (witness program versions 1 through 16)
+
+  These script pub key forms include only standard forms accepted by the Bitcoin network, which ensures the resulting transaction will propagate to miners.
```
Comment (ariard) [inline Protocol.md]: I would phrase this differently, "accepted by the wider set of deployed Bitcoin clients in the network, which increase the chances of successful propagation to miners". Policy rules are heterogeneous by design.

**Review by Tibo-lg — APPROVED** [review]

File: Protocol.md (hunk 1 above)
Comment (benthecarman) [reply Protocol.md:277]: Added to apply to funding inputs

Comment (ariard) [reply Protocol.md:277]: Ooops, my bad, I meaned that _input_ scriptpubkey standardness isn't the same that _output_ one. Non-Taproot and v1+ witness program are excluded for inputs. So I would formalize differently,
"If the scriptpubkey is sent to: P2PKH, P2SH, P2WPH, P2WSH, segwit v1+"
"If the scriptpubkey is spent from : P2WPKH, P2WSH"

Note the restriction on inputs excluding P2SH, P2PKH, THIS IS UNSAFE TO USE NON-SEGWIT INPUTS !

A malicious counterparty can malleate the input, confirm the funding, break the sigs of second-stage dlcs/refund and thus take in hostage collateral funds.

Comment (benthecarman) [reply Protocol.md:277]: Fixed to use the same wording for inputs as the BOLTs

**Review by ariard — COMMENTED** [review]
SGTM 2256f71 :)

**Pull request merged**